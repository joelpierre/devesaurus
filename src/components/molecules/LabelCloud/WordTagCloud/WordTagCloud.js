import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import { StaticQuery, graphql } from 'gatsby';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import styles from '../LabelCloud.module.scss';
import Label from '../../../atoms/Label/Label';
import { mapTaxonomyIcon, mapTaxonomyTheme } from '../../../../utils';

export const PureWordTagCloudList = ({ allWordpressWpWordTag }) => (
  <>
    {allWordpressWpWordTag.edges.map(({ node }) => (
      <li
        data-test="word-tag-cloud-item"
        key={node.id}
        className={classNames(styles['label-cloud__item'])}
      >
        <Label
          className={styles['label-cloud__label']}
          link={'/devinitions/tag/' + node.slug}
          theme={mapTaxonomyTheme(node.slug)}
        >
          <FontAwesomeIcon
            icon={mapTaxonomyIcon(node.slug)}
            className={styles['label-cloud__icon']}
          />
          {node.name}
        </Label>
      </li>
    ))}
  </>
);

const WordTagCloud = ({ className }) => (
  <ul
    data-test="component-word-tag-cloud"
    className={classNames(styles['label-cloud'], className)}
  >
    <StaticQuery
      data-test="component-word-tag-cloud-query"
      query={query}
      render={props => <PureWordTagCloudList {...props} />}
    />
  </ul>
);

WordTagCloud.defaultProps = {
  className: undefined,
};

WordTagCloud.propTypes = {
  className: PropTypes.string,
};

export default WordTagCloud;

const query = graphql`
  {
    allWordpressWpWordTag {
      edges {
        node {
          id
          name
          slug
          taxonomy
        }
      }
    }
  }
`;
