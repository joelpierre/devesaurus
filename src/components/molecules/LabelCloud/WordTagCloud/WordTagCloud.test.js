import React from 'react';
import { shallow } from 'enzyme/build';

import WordTagCloud, { PureWordTagCloudList } from './WordTagCloud';
import {
  checkProps,
  findByTestAttr,
  matchSnapshot,
} from '../../../../utils/test';

const defaultProps = {};
const defaultPureProps = {
  allWordpressWpWordTag: {
    edges: [
      {
        node: {
          id: 1,
          slug: 'test-slug',
          name: 'test-name',
          taxonomy: 'test-tax',
        },
      },
    ],
  },
};

/**
 * Factory function to create a ShallowWrapper for the WordTagCloud component.
 * @param {object} props - Component props specific to setup
 * @returns {ShallowWrapper}
 */
const setup = (props = {}) => {
  const setupProps = { ...defaultPureProps, ...props };
  return shallow(<WordTagCloud {...setupProps} />);
};
/**
 * Factory function to create a ShallowWrapper for the PureWordTagCloud component.
 * @param {object} props - Component props specific to setup
 * @returns {ShallowWrapper}
 */
const pureSetup = (props = {}) => {
  const setupProps = { ...defaultPureProps, ...props };
  return shallow(<PureWordTagCloudList {...setupProps} />);
};

describe('<PureWordTagCloudList/>', () => {
  let wrapper;
  let pureWrapper;

  beforeEach(() => {
    wrapper = setup();
    pureWrapper = pureSetup();
  });

  it('renders the WordTagCloudList Component without errors', () => {
    const component = findByTestAttr(wrapper, 'component-word-tag-cloud');
    expect(component.length).toBe(1);
  });

  it('renders the WordTagCloudList Query without errors', () => {
    const component = findByTestAttr(wrapper, 'component-word-tag-cloud-query');
    expect(component.length).toBe(1);
  });

  it('renders the PureWordTagCloudList Component without errors', () => {
    const component = findByTestAttr(pureWrapper, 'word-tag-cloud-item');
    expect(component.length).toBe(1);
  });

  it('should render with correct props', () => {
    checkProps(PureWordTagCloudList, defaultProps);
  });

  it('should match snapshot PureWordTagCloudList', () => {
    matchSnapshot(pureWrapper);
  });

  it('should match snapshot WordTagCloudList', () => {
    matchSnapshot(wrapper);
  });
});
