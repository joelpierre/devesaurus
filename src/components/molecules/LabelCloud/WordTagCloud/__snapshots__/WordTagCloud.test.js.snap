// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PureWordTagCloudList/> should match snapshot PureWordTagCloudList 1`] = `
<Fragment>
  <li
    className="label-cloud__item"
    data-test="word-tag-cloud-item"
    key="1"
  >
    <Label
      caps={false}
      classes="label-cloud__label"
      link="/word-tag/test-slug"
      size="sm"
      theme="brand"
    >
      <FontAwesomeIcon
        border={false}
        className="label-cloud__icon"
        fixedWidth={false}
        flip={null}
        icon={
          Array [
            "far",
            "atom-alt",
          ]
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        title=""
        transform={null}
      />
      test-name
    </Label>
  </li>
</Fragment>
`;

exports[`<PureWordTagCloudList/> should match snapshot WordTagCloudList 1`] = `
<ul
  className="label-cloud"
  data-test="component-word-tag-cloud"
>
  <mockConstructor
    data-test="component-word-tag-cloud-query"
    render={[Function]}
  />
</ul>
`;
